lib/sycamore/sycamore/llms/openai.py

```diff
@@ -250,21 +254,19 @@ def __init__(
     # The actual openai client is not pickleable, This just says to pickle the wrapper, which can be used to
     # recreate the client on the other end.
     def __reduce__(self):
-        def deserializer(kwargs):
-            return OpenAI(**kwargs)
 
-        kwargs = {"client_wrapper": self.client_wrapper, "model_name": self._model_name}
+        kwargs = {"client_wrapper": self.client_wrapper, "model_name": self._model_name, "cache": self._cache}
 
-        return deserializer, (kwargs,)
+        return openai_deserializer, (kwargs,)
```

Comment:
Just out of curiosity, what changed with the deserializer?

Reply:
No logical change, pickle just seems to complain if the function is part of the class you're serializing (based on the test in the description)